compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEOpUtils.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_AMD_AIE_TRANSFORMS_AMDAIEOPUTILS_H_
+#define IREE_AMD_AIE_TRANSFORMS_AMDAIEOPUTILS_H_
+
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/IR/Block.h"
+#include "mlir/IR/Operation.h"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+/// Return a vector of the parent operations that are of type 'OpTy'
+template <typename OpTy>
+SmallVector<OpTy> getParentsOfTypeIncluding(Operation *op) {
```

Comment:
This function name is not clear to me. Maybe rename it to `getOpTyParentOp` or something?

Reply:
This function is named after `Operation::getParentOfType`: https://mlir.llvm.org/doxygen/classmlir_1_1Operation.html#a9968b118b25a099be92400e88239153c, and I think it should stay consistent with that one. Is the `including` part unclear?

Reply:
Renamed it to `getInclusiveParentsOfType` and explained it in the comment.

Reply:
I think this is better. Thanks!